src/vs/editor/browser/viewParts/viewCursors/viewCursor.ts

```diff
@@ -128,6 +129,17 @@ export class ViewCursor {
 			this._positionTop = visibleRange.top;
 			this._positionLeft = visibleRange.left;
 			this._isInViewport = true;
+
+			let visibleRangeForCharacter = ctx.linesVisibleRangesForRange({
```

Comment:
@rebornix This extra call should be done only if the cursor style is a block cursor